src/XML/src/FMI3/fmi3_xml_parser.c

```diff
@@ -743,13 +765,23 @@ int fmi3_xml_set_attr_array(fmi3_xml_parser_context_t *context, fmi3_xml_elm_enu
     }
 
     /* write all attributes to array of correct type */
-    if (fmi3_xml_str_to_array_floatXX(context, str, arrPtr, arrSize, primType)) {
+    if (fmi3_xml_str_to_array(context, str, arrPtr, arrSize, primType)) {
         jm_string elmName = fmi3_element_handle_map[elmID].elementName;
         jm_string attrName = fmi3_xmlAttrNames[attrID];
-        fmi3_xml_parse_error(context, "XML element '%s': could not parse value for float attribute '%s'='%s'", elmName, attrName, str);
+        char* type_as_str = "";
+        if (fmi3_base_type_enu_is_float(primType->baseType)) {
+            type_as_str = "float";
+        } else if(fmi3_base_type_enu_is_int(primType->baseType)) {
+            type_as_str = "int";
+        }
+        fmi3_xml_parse_error(context, "XML element '%s': could not parse value for %s attribute '%s'='%s'",
+            elmName,
+            type_as_str,
+            attrName,
+            str
+        );
```

Comment:
Here we have two args on same line, then one for each row? And sometimes ); on new line and sometimes not. I think we need to discuss how to format function calls.